### ISSUE TITLE
The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI

### ISSUE THREAD

**Issue by metma99** [aws/aws-parallelcluster#3795]
**Required Info:**
- AWS ParallelCluster version [e.g. 2.9.0]: 2.9.1
- Full cluster configuration without any credentials or personal data:
[aws]
aws_region_name = ....

  [aliases]
  ssh = ssh {CFN_USER}@{MASTER_IP} {ARGS}

  [global]
  cluster_template = default
  update_check = true
  sanity_check = true

  [cluster default]
  key_name = ...
  base_os = alinux2
  scheduler = slurm
  master_instance_type = t3a.large
  vpc_settings = default
  queue_settings = compute
  efs_settings = default
  scaling_settings = default
  post_install = ...
  ec2_iam_role = ...
  tags = ...

  [vpc default]
  vpc_id = ...
  master_subnet_id = ...
  use_public_ips = false

  [queue compute]
  enable_efa = false
  compute_resource_settings = default
  compute_type = spot

  [compute_resource default]
  instance_type = c5a.4xlarge
  max_count = 25

  [efs default]
  shared_dir = efs
  encrypted = false
  performance_mode = generalPurpose

  [scaling default]
  scaledown_idletime = 15

- Cluster name: if really required I can provide it

**Bug description and how to reproduce:**
  Note this cluster setup has worked before. I used it 6 months ago w/o any issues.
  Yesterday I was not able to run any jobs on the cluster and finally tracked the issue down to:
  I deleted the cluster successfully and issued:
  pcluster create omeros-nextflow-pcluster

  This is the error message:
  pcluster create omeros-nextflow-pcluster
  Beginning cluster creation for cluster: omeros-nextflow-pcluster
  ERROR: Unable to validate configuration parameters. Please double check your cluster configuration.
  The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI. Specify an   instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images'   to discover the architecture of the instance type or AMI.
  I checked describe-instance-types, describe-images all fine.
  The above specified instance types are available for architecture x86_64. The image is also x86_64.

  1) aws ec2 describe-instance-types --filters Name=processor-info.supported-architecture,Values=x86_64 --query "InstanceTypes[*].InstanceType" --output text

  The instance types specified in config are indeed of architecture x86_64 (not showing the rather lengthy output):
  master_instance_type = t3a.large
  instance_type = c5a.4xlarge

  2) aws ec2 describe-images --region us-west-2 --image-ids ami-09975e5a942d87be0
  output:
{
    "Images": [
        {
            "Architecture": "x86_64",
            "CreationDate": "2020-09-14T23:26:13.000Z",
            "ImageId": "ami-09975e5a942d87be0",
            "ImageLocation": "amazon/aws-parallelcluster-2.9.1-amzn2-hvm-x86_64-202009142226",
            "ImageType": "machine",
            "Public": true,
            "OwnerId": "247102896272",
            "PlatformDetails": "Linux/UNIX",
            "UsageOperation": "RunInstances",
            "State": "available",
            "BlockDeviceMappings": [
                {
                    "DeviceName": "/dev/xvda",
                    "Ebs": {
                        "DeleteOnTermination": true,
                        "SnapshotId": "snap-08f7546394e9eeffb",
                        "VolumeSize": 25,
                        "VolumeType": "gp2",
                        "Encrypted": false
                    }
                }
            ],
            "EnaSupport": true,
            "Hypervisor": "xen",
            "ImageOwnerAlias": "amazon",
            "Name": "aws-parallelcluster-2.9.1-amzn2-hvm-x86_64-202009142226",
            "RootDeviceName": "/dev/xvda",
            "RootDeviceType": "ebs",
            "SriovNetSupport": "simple",
            "VirtualizationType": "hvm" 

**If you are reporting issues about scaling or job failure:**

We cannot work on issues without proper logs. We **STRONGLY** recommend following [this guide](https://github.com/aws/aws-parallelcluster/wiki/Creating-an-Archive-of-a-Cluster's-Logs) and attach the complete cluster log archive with the ticket.

  This script failed issuing an error that files are missing:
  ERROR:retrieve-cluster-logs.py:Unable to find log group in region us-west-2 for cluster 

For issues with AWS ParallelCluster >= v2.9.0 and scheduler == slurm, please attach the following logs:
* From Head node: `/var/log/parallelcluster/clustermgtd`, `/var/log/parallelcluster/clusterstatusmgtd` (if version >= 3.2.0), `/var/log/parallelcluster/slurm_resume.log`, `/var/log/parallelcluster/slurm_suspend.log`, `/var/log/parallelcluster/slurm_fleet_status_manager.log` (if version >= 3.2.0) and`/var/log/slurmctld.log` 
* From Compute node:  `/var/log/parallelcluster/computemgtd.log`, and `/var/log/slurmd.log`
Otherwise, please attach the following logs:
* From Head node: `/var/log/jobwatcher`, `/var/log/sqswatcher`, and `/var/log/slurmctld.log` if scheduler == slurm.
* From Compute node:`/var/log/nodewatcher`, and `/var/log/slurmd.log` if scheduler == slurm

  I have no access to any of those files above. Instead I am posting:
[pcluster-cli_02182022.log](https://github.com/aws/aws-parallelcluster/files/8100021/pcluster-cli_02182022.log)

**If you are reporting issues about cluster creation failure or node failure:**
If the cluster fails creation, please re-execute `create` action using `--norollback` option.
If I understood you correctly then I did the following resulting in the same error.
pcluster create omeros-nexflow-pcluster --norollback

We cannot work on issues without proper logs. We **STRONGLY** recommend following [this guide](https://github.com/aws/aws-parallelcluster/wiki/Creating-an-Archive-of-a-Cluster's-Logs) and attach the complete cluster log archive with the ticket.

* From Head node: `/var/log/cloud-init.log`, `/var/log/cfn-init.log`, and `/var/log/chef-client.log`
* From Compute node:  `/var/log/cloud-init-output.log`

  Only the cloud-init.log file was available which is attached:
  [cloud-init.log](https://github.com/aws/aws-parallelcluster/files/8100025/cloud-init.log)

I tried to include as much info as possible. Kindly let me know if you require any more details.
Thank you very much in advance for any input you might have.

Markus

**Comment by hanwen-pcluste** [comment]
Hi metama,

This problem and the workaround is documented in this [Wiki](https://github.com/aws/aws-parallelcluster/wiki/Configuration-validation-failure:-architecture-of-AMI-and-instance-type-does-not-match)

If you are creating a new cluster, we would recommend using the latest versions of pcluster (2.11.4 or 3.1.1)

Thank you,
Hanwen

**Comment by metma99** [comment]
Dear Hanwen:
Thank you very much for your answer.
I did what you suggested, the creation continues but other error messages pop up eventually leading to failure.
Please advice if I should close this issue and open another one.
In the meantime I am adding my steps here:

1) edit config file and set sanity_check = false
2) pcluster delete {clustername}
3) pcluster create {clustername}
Beginning cluster creation for cluster: {clustername}
ERROR: Unable to validate configuration parameters. Please double check your cluster configuration.
The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI. Specify an instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images' to discover the architecture of the instance type or AMI.
ERROR: Unable to validate configuration parameters. Please double check your cluster configuration.
The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI. Specify an instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images' to discover the architecture of the instance type or AMI.
Info: There is a newer version 3.1.1 of AWS ParallelCluster available.
Creating stack named: parallelcluster-{clustername}
Status: ComputeSecurityGroup - CREATE_IN_PROGRESS
Status: {clustername} - ROLLBACK_IN_PROGRESS
Cluster creation failed.  Failed events:
   - AWS::CloudFormation::Stack MasterServerSubstack Embedded stack arn:aws:cloudformation:us-west-2:760688881864:stack/{clustername}-09X9/0c10db10-9664-11ec-80bd-06886d5ee2a1 was not successfully created: The following resource(s) failed to create: [MasterServer].
    - AWS::CloudFormation::Stack {clustername}-MasterServerSubstack-Q2FPJR6309X9 The following resource(s) failed to create: [MasterServer].
    - AWS::EC2::Instance MasterServer Received FAILURE signal with UniqueId i-08a2b94aefaf7ae80
3) pcluster list
   {clustername} ROLLBACK_IN_PROGRESS  2.9.1
4) pcluster delete {clustername}
5) pcluster list
   no output
6) pcluster create {clustername} --norollback # obtained from here https://github.com/aws/aws-parallelcluster/issues/1315
Beginning cluster creation for cluster: {clustername}
ERROR: Unable to validate configuration parameters. Please double check your cluster configuration.
The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI. Specify an instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images' to discover the architecture of the instance type or AMI.
ERROR: Unable to validate configuration parameters. Please double check your cluster configuration.
The architecture 'x86_64' of the specified instance type does not match the architecture 'arm64' of the specified AMI. Specify an instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images' to discover the architecture of the instance type or AMI.
Info: There is a newer version 3.1.1 of AWS ParallelCluster available.
Creating stack named: {clustername}
Status: {clustername} - CREATE_FAILED
Cluster creation failed.  Failed events:
   - AWS::CloudFormation::Stack {clustername} The following resource(s) failed to create: [MasterServerSubstack].
   - AWS::CloudFormation::Stack MasterServerSubstack Embedded stack arn:aws:cloudformation:us-west-2:760688881864:stack/{clustername}-MasterServerSubstack-TZQ7E4Q0KUAX/6abde230-9672-11ec-bf2c-02f8b22b3379 was not successfully created: The following resource(s) failed to create: [MasterServer].
    - AWS::CloudFormation::Stack {clustername}-MasterServerSubstack-TZQ7E4Q0KUAX The following resource(s) failed to create: [MasterServer].
    - AWS::EC2::Instance MasterServer Received FAILURE signal with UniqueId i-09a6d3c42d934e751
7) pcluster list
{clustername} CREATE_FAILED  2.9.1
8) pcluster ssh {clustername}
I cannot find any helpful log files as suggested here:
https://docs.aws.amazon.com/parallelcluster/latest/ug/troubleshooting.html

Does anybody have any idea what is going on?
Many thanks in advance,
Markus

**Comment by chenwany** [comment]
Hi, seems the headnode create is failed. After the cluster creation failed, resources will be delete. So you saw `ROLLBACK_IN_PROGRESS `
Can you create the cluster with `--norollback` option, so the stack will not be deleted after creation failure. Then you can pcluster ssh {clustername} to find logs files in headnode.
Are you able to find something interesting in logs from Head node: /var/log/cloud-init.log, /var/log/cfn-init.log, and /var/log/chef-client.log ?

**Comment by metma99** [comment]
Dear all:
First of all thank you very much to chenway and hanwen.
Secondly, the solution is not related to pcluster but I thought I post it here and close the issue.

In a nutshell this is how my issue got resolved:
a) first upgrading to pcluster 2.11.4 resolved the issue with the architecture mismatch.
b) this created another issue: I was not able to start the master. Based on the eror message 
"INFO - utils -     - AWS::CloudWatch::Dashboard HeadNodeDashboard User: arn:aws:iam::76068888186
4:user/...-admin is not authorized to perform: cloudwatch:GetDashboard on resource: arn:aws:cloudwatch::760688881864:da shboard/parallelcluster-.... because no identity-based policy allows the cloudwatch:GetDashb oard action (Service: AmazonCloudWatch; Status Code: 403; Error Code: AccessDenied; Request ID: 7070f07f-03ee-48ce-adfa-f806ca4be04d; Proxy: null)"
Based on this we added to the cloudwatch:GetDashboard the line "cloudwatch:Get*"
c) This raised another error:
"    - AWS::CloudWatch::Dashboard HeadNodeDashboard User: arn:aws:iam::760688881864:user/....admin is not authorized to perform: cloudwatch:PutDashboard on resource: ..."
This error got resolved with creating a custom policy CloudWatch-allows:
Custom policy we created, “...-CloudWatch-allows”

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "cloudwatch:PutDashboard",
            "Resource": "*"
        }
    ]
}

d) viola - I now cann access the master again.
Thank you to everybody who helped me out!
Markus
